app/models/site.rb

```diff
@@ -62,12 +62,17 @@ def lon=(val)
     end
   end
 
+  # Returns true if the site geometry is null or has type 'ST_Point'.
+  def point?
+    geometry.nil? || geometry.geometry_type.type_name == 'Point'
+  end
+
```

Comment:
Makes perfect sense. You addressed what was essentially the subtext of my question - "is the name misleading?"